File: Java/2264. Largest 3-Same-Digit Number in String.java
class Solution {
    public String largestGoodInteger(String num) {
        int start = 0, ans = -1;

        for (int i = 1; i < num.length(); i++) {
            if (num.charAt(i) != num.charAt(i-1)) {
                start = i;
            }

            if (i - start == 2) {
                start = i;
                ans = Math.max(ans, (int)num.charAt(i-1));
            }
        }

        return ans == -1 ? "" : new String(new char[]{(char)ans, (char)ans, (char)ans});
    }
}

File: Python3/2264. Largest 3-Same-Digit Number in String.py
class Solution:
    def largestGoodInteger(self, num: str) -> str:
        start = 0
        ans = -1

        for i, n in enumerate(num[1:], 1):
            # start a new count
            if n != num[i-1]:
                start = i

            if i - start == 2:
                ans = max(ans, int(num[i-1]))
                # need to update the start pointer
                start = i
                
        return str(ans) * 3 if ans != -1 else ''